plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsSecurityContext.java

```diff
@@ -72,6 +76,13 @@ class HdfsSecurityContext {
             // 7) allow code to initiate kerberos connections as the logged in user
             // Still far and away fewer permissions than the original full plugin policy
         };
+        WEBHDFS_PERMISSIONS = new Permission[] {
+            new AuthPermission("doAs")
+        };
+        SWEBHDFS_PERMISSIONS = new Permission[] {
+            new AuthPermission("doAs"),
+            new RuntimePermission("setFactory"),
```

Comment:
This might be a bit in the weeds as a request, but the above registry of permissions details why they are needed. Can we add a quick blurb for each permission about why they are included?

Reply:
Sure, I've added comments to that section. For some more details:

- `doAs` permission was required (IIRC) due to the call [here](https://github.com/trakos/hadoop/blob/87e39011908cd63627beeab0b68ef89a2ac72454/hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/web/WebHdfsFileSystem.java#L702-L702). This permissions was already required for HDFS plugin in Elasticsearch, but only for Kerberized setup. Now, it's also required for webhdfs, even without Kerberos.
- `setFactory` is required for the call [here](https://github.com/trakos/hadoop/blob/87e39011908cd63627beeab0b68ef89a2ac72454/hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/web/SSLConnectionConfigurator.java#L61-L61). This is required only for the `swebhdfs` scheme, as `webhdfs` doesn't use SSL connection.